torch/_higher_order_ops/utils.py

```diff
@@ -546,3 +538,62 @@ def validate_subgraph_args_types(lifted_args: Union[tuple[Any, ...], list[Any]])
     assert all(
         isinstance(arg, (torch.Tensor, int, torch.SymInt)) for arg in lifted_args
     ), f"{lifted_args} can only be of {allowed_types} but got {tuple(type(arg) for arg in lifted_args)}"
+
+
+def check_input_alias_and_mutation(
+    gm: torch.fx.GraphModule,
+    fake_args: List[FakeTensor],
+) -> Tuple[List[int], dict[int, int], dict[int, int]]:
+    with disable_proxy_modes_tracing():
+        """This function creturns mutated inputs, inp-out alias and out-out alias
+        in the graph module gm.
+        """
+
+        def _tensor_version(t) -> Optional[int]:
+            if isinstance(t, FakeTensor):
+                return t._version
+            return None
```

Comment:
Can you assert that all Tensors are FakeTensors? Otherwise this function is wrong